routing/bicycle_directions.cpp

```diff
@@ -170,7 +185,13 @@ void BicycleDirectionsEngine::Generate(IRoadGraph const & graph, vector<Junction
   Route::TTimes turnAnnotationTimes;
   Route::TStreets streetNames;
 
-  MakeTurnAnnotation(resultGraph, delegate, routeGeometry, turns, turnAnnotationTimes, streetNames);
+  MakeTurnAnnotation(resultGraph, delegate, routeGeometry, turns, turnAnnotationTimes,
+                     streetNames, routeSegs);
+  // @TODO(bykoianko) The invariant below it's an issue but now it's so and it should be checked.
+  // The problem is every edge is added as a pair of points to route geometry.
+  // So all the points except for beginning and ending are duplicated. It should
+  // be fixed in the future.
+  ASSERT_EQUAL(routeGeometry.size(), 2 * routeSegs.size(), ());
```

Comment:
Please, replace this ASSERTion by two:
```c++
ASSERT_EQUAL(routeGeometry.size(), 2 * (pathSize - 2) + 2, ());
ASSERT_EQUAL(routeSegs.size(), pathSize - 1, ());
```
with the comment explaining that all internal points of path are duplicated in routeGeometry, and that number of routeSegs should be equal to number of edges in the path.